airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -87,11 +117,28 @@ public OAuthConsentRead getDestinationOAuthConsent(final DestinationOauthConsent
     final Map<String, Object> metadata = generateDestinationMetadata(destinationDefinitionIdRequestBody.getDestinationDefinitionId());
     final OAuthConsentRead result;
     if (OAuthConfigSupplier.hasOAuthConfigSpecification(spec)) {
+      JsonNode oAuthInputConfigurationForConsent;
+
+      if (destinationDefinitionIdRequestBody.getDestinationId() == null) {
+        oAuthInputConfigurationForConsent = destinationDefinitionIdRequestBody.getoAuthInputConfiguration();
+      } else {
+        final SourceConnection hydratedSourceConnection =
```

Comment:
Shouldn't this be a `DestinationConnection`?

Reply:
Done